From a5ea6ff7ff301d739d3dc8145db1578f504ee4ca Mon Sep 17 00:00:00 2001
From: fuscated <fuscated@2a5c6006-c6dd-42ca-98ab-0921f2732cef>
Date: Tue, 16 Mar 2021 23:28:01 +0000
Subject: [PATCH] - openfilelist: Fix build with GCC-11

> It seems this new version of GCC would require the predicate to have a
  const operator().
> But I suppose there is no need to reinvent std::less, so use it instead.

git-svn-id: https://svn.code.sf.net/p/codeblocks/code/trunk@12303 2a5c6006-c6dd-42ca-98ab-0921f2732cef
---
 src/plugins/openfileslist/openfileslistplugin.h | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/src/plugins/openfileslist/openfileslistplugin.h b/src/plugins/openfileslist/openfileslistplugin.h
index c59d67ba1..91cfc94cb 100644
--- a/src/plugins/openfileslist/openfileslistplugin.h
+++ b/src/plugins/openfileslist/openfileslistplugin.h
@@ -10,6 +10,7 @@
 #include <projectfile.h>
 
 #include <wx/dynarray.h>
+#include <functional>
 
 class wxTreeCtrl;
 class wxTreeEvent;
@@ -20,12 +21,8 @@ class EditorBase;
 struct TargetFilesData
 {
     TargetFilesData() : activeFile(nullptr) {} // ctor
-    // Functor for the std::set predicate to sort the opened editor files according to their tab order
-    struct compareLess
-    {
-        bool operator()(const ProjectFile* lhs, const ProjectFile* rhs) { return lhs->editorTabPos < rhs->editorTabPos; }
-    };
-    typedef std::set<ProjectFile*, compareLess> OpenFilesSet;
+
+    typedef std::set<ProjectFile*, std::less<ProjectFile*>> OpenFilesSet;
     ProjectFile* activeFile;
     OpenFilesSet openFiles;
 };
